File: cs/src/AutoDiff/ConstraintUtility.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace AutoDiff
{
    /// <summary>
    /// Represents a constraint utility function.
    /// </summary>
    [Serializable]
    public class ConstraintUtility : Term
    {
        /// <summary>
        /// Constructs a new instance of the <see cref="LTConstraint"/> type.
        /// </summary>
        /// <param name="constraint">The constraint term</param>
        /// <param name="utility">The utility term</param>        
        public ConstraintUtility(Term constraint, Term utility)
        {
            Constraint = constraint;
            Utility = utility;			
        }

        /// <summary>
        /// Gets the constraint.
        /// </summary>
        public Term Constraint { get; private set; }

        /// <summary>
        /// Gets the utility.
        /// </summary>
        public Term Utility { get; private set; }
		
		
        /// <summary>
        /// Accepts a term visitor
        /// </summary>
        /// <param name="visitor">The term visitor to accept</param>
        public override void Accept(ITermVisitor visitor)
        {
            visitor.Visit(this);
        }

        /// <summary>
        /// Accepts a term visitor with a generic result
        /// </summary>
        /// <typeparam name="TResult">The type of the result from the visitor's function</typeparam>
        /// <param name="visitor">The visitor to accept</param>
        /// <returns>
        /// The result from the visitor's visit function.
        /// </returns>
        public override TResult Accept<TResult>(ITermVisitor<TResult> visitor)
        {
            return visitor.Visit(this);
        }
		public override Term Negate ()
		{
			throw new Exception("Do not negate a Constraint Utility");
			//return this;
		}
		
		public override Term AggregateConstants ()
		{
			Constraint = Constraint.AggregateConstants ();
			Utility = Utility.AggregateConstants ();
			return this;
		}
		public override Term Derivative (Variable v)
		{
			throw new Exception ("Symbolic Derivation of ConstraintUtility not supported.");
		}
		
		public override string ToString ()
		{
			return string.Format ("[ConstraintUtility: Constraint={0}, Utility={1}]", Constraint, Utility);
		}
    }
}

File: java/src/main/java/AutoDiff/ConstraintUtility.java
//
// Translated by CS2J (http://www.cs2j.com): 12.11.2016 09:40:33
//

package AutoDiff;

/**
* Represents a constraint utility function.
*/
public class ConstraintUtility  extends Term 
{
    /**
    * Constructs a new instance of the 
    *  {@link #LTConstraint}
    *  type.
    * 
    *  @param constraint The constraint term
    *  @param utility The utility term
    */
    public ConstraintUtility(Term constraint, Term utility) throws Exception {
        setConstraint(constraint);
        setUtility(utility);
    }

    /**
    * Gets the constraint.
    */
    private Term __Constraint;
    public Term getConstraint() {
        return __Constraint;
    }

    public void setConstraint(Term value) {
        __Constraint = value;
    }

    /**
    * Gets the utility.
    */
    private Term __Utility;
    public Term getUtility() {
        return __Utility;
    }

    public void setUtility(Term value) {
        __Utility = value;
    }

//    /**
//    * Accepts a term visitor
//    *
//    *  @param visitor The term visitor to accept
//    */
//    public void accept(ITermVisitor visitor) throws Exception {
//        visitor.visit(this);
//    }

    /**
    * Accepts a term visitor with a generic result
    * The type of the result from the visitor's function
    *  @param visitor The visitor to accept
    *  @return 
    * The result from the visitor's visit function.
    */
    public <TResult>TResult accept(ITermVisitor<TResult> visitor) throws Exception {
        return visitor.visit(this);
    }

    public Term negate() throws Exception {
        throw new Exception("Do not negate a Constraint Utility");
    }

    //return this;
    public Term aggregateConstants() throws Exception {
        setConstraint(getConstraint().aggregateConstants());
        setUtility(getUtility().aggregateConstants());
        return this;
    }

    public Term derivative(Variable v) throws Exception {
        throw new Exception("Symbolic Derivation of ConstraintUtility not supported.");
    }

    public String toString() {
        try
        {
            return String.format("[ConstraintUtility: Constraint=%s, Utility=%s]", getConstraint(), getUtility());
        }
        catch (RuntimeException __dummyCatchVar0)
        {
            throw __dummyCatchVar0;
        }
        catch (Exception __dummyCatchVar0)
        {
            throw new RuntimeException(__dummyCatchVar0);
        }
    
    }

}


